File: java/service/src/main/java/service/ParticipantService.java
package service;

import interfaces.ParticipantRepository;
import models.Participant;
import models.Rally;
import models.Team;

public class ParticipantService {
    private ParticipantRepository participantRepository;

    public ParticipantService(ParticipantRepository participantRepository) {
        this.participantRepository = participantRepository;
    }

    public Participant addParticipant(Team participantTeam, Rally rallyParticipatesTo, String participantName) {
        Participant participant = new Participant(participantTeam, rallyParticipatesTo, participantName);
        return participantRepository.save(participant);
    }

    public Iterable<Participant> getAllMembersOfTeam(String teamName) {
        return participantRepository.findMembersOfTeam(teamName);
    }
}

File: csharp/Business/Services/Implementation/ParticipantService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


using Repository.Interfaces;
using Domain.Models;

namespace Business.Services
{
    public class ParticipantService
    {
        private IParticipantRepository participantRepository;

        public ParticipantService(IParticipantRepository participantRepository)
        {
            this.participantRepository = participantRepository;
        }

        public Participant AddParticipant(Team participantsTeam, Rally rallyParticipatesTo, String participantName)
        {
            Participant participant = new Participant(rallyParticipatesTo, participantsTeam, participantName);
            return participantRepository.Add(participant);
        }

        public ICollection<Participant> GetAllMembersOfTeam(String teamName)
        {
            return participantRepository.GetMembersOfTeam(teamName);
        }

        public Participant? GetParticipantByName(String participantName)
        {
            return participantRepository.GetByName(participantName);
        }
    }
}
